lib/cache/timeout-manager.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * A timer cache manager module responsible for processing timers and cleaning up uncompleted lookups.
+ * @module timeout-manager
+ * @private
+ */
+
+var cacheManager = require('cache-manager');
+// to cache timers, must be an in memory store.
+var timerStore = cacheManager.caching({store: 'memory'});
+var buildKey = require('./key-builder').buildTimerCacheKey;
+var lnManager = require('./lookup-numbers-manager');
+var keyManager = require('./key-manager');
+var delKey = require('./utils').delKey;
+var loggingFunc = require('../logging/logger-message-formatter').getLoggerMessageFunc('timeout-manager');
+var getLogger = require('../logging/logger-options-parser').getLoggerFromOptions;
+
+/**
+ * Remove the timer from the internal memory cache
+ * @param {string} uniqueId unique id used to compute cache key
+ * @param {object} cache the cache itself
+ * @returns {Promise} with void result and rejects if operation failed
+ */
+function removeTimerFromCache(uniqueId, cache) {
+  try {
+    var timerKey = buildKey(uniqueId);
+    return delKey(cache, timerKey);
+  } catch (e) {
+    return Promise.reject(e);
+  }
+}
+
+/**
+ * Gets the timer from cache
+ * @param {string} uniqueId unique id used to compute cache key
+ * @param {object} cache the cache itself
+ * @returns {Promise} whose result is the timer object
+ */
+function getKeyFormCache(uniqueId, cache){
```

Comment:
I think we should either rename `removeTimerFromCache` to `removeKeyFromCache` or rename `getKey` and `setKey` to `getTimer` and `setTimer`, for consistency.